### ISSUE TITLE
[Bug] When Batch Size > 1 precision has declined

### ISSUE THREAD

**Issue by Prayer3th** [sgl-project/sglang-jax#116]
### Checklist

- [x] 1. I have searched related issues but cannot get the expected help.
- [x] 2. The bug has not been fixed in the latest version.
- [x] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [x] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sgl-jax/discussions/new/choose Otherwise, it will be closed.
- [x] 5. Please use English, otherwise it will be closed.

### Describe the bug

**temperature = 0, dataset = gsm8k**
* batch size = 1
The precision can be fully aligned after multiple tests
* batch size = 16 
The precision drops by 2.5% 

### Reproduction

**launch command**
JAX_COMPILATION_CACHE_DIR=/tmp/jit_cache  python3 -u -m sgl_jax.launch_server --model-path Qwen/Qwen3-8B --trust-remote-code  --dist-init-addr=0.0.0.0:10011 --nnodes=1  --tp-size=4 --device=tpu --random-seed=3 --node-rank=0 --mem-fraction-static=0.8 --chunked-prefill-size=8192 --download-dir=/tmp --dtype=bfloat16 --jax-precompile-decode-bs-paddings 16 --max-running-requests 16 --max-total-tokens 257536 --skip-server-warmup --attention-backend=fa --jax-precompile-prefill-token-paddings 8192 --page-size=64 --enable-precision-tracer --disable-radix-cache

**test command**
evalscope eval  --model Qwen3-8B --api-url http://127.0.0.1:30000/v1 --api-key EMPTY --eval-type service --datasets gsm8k --eval-batch-size 16 --limit 50

### Environment

google tpu v6e * 4
